src/main/java/com/esri/samples/localserver/local_server_geoprocessing/LocalServerGeoprocessingController.java

```diff
@@ -162,11 +160,18 @@ protected void handleClearResults() {
   }
 
   /**
-   * Stops the animation and disposes of application resources.
+   * Stops and releases all resources used in application.
    */
   void terminate() {
     if (mapView != null) {
       mapView.dispose();
     }
+
+    // stop local server when jvm shuts down
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {

```

Comment:
The local server is watching the Java process and should shutdown when the JVM stops so doing this in a shutdown hook shouldn't be necessary.